[PC-1301] - Portenta C33 User Manual and Cloud IoT (#1123)
arduino/docs-content: content/hardware/04.pro/boards/portenta-c33/tutorials/user-manual/content.md

- **Public key import**: The public key derived from the generated private key is imported to the secure element with key ID 899.
- **Signature verification**: The example code then verifies the signature using the imported public key and prints a success or failure message accordingly.

## Arduino IoT Cloud 

The Portenta C33 is fully compatible with the Arduino Cloud IoT, making it easy to develop professional IoT applications fast. By using the IoT Cloud, you can, for example, monitor sensor data, control your board and actuators connected to it remotely, and update your device's firmware remotely also. 

> ***To use the Arduino IoT Cloud, you need an account. If you do not have an account, create one for free [here](https://cloud.arduino.cc/). To use the Arduino Web Editor or Arduino IoT Cloud, the Arduino Create Agent must be running on your computer. You can install the Arduino Create Agent [here](https://create.arduino.cc/getting-started/plugin/welcome).***

Author: I agree with you. Change done.

![Arduino IoT Cloud Setup Device pop-up window](assets/user-manual-17.png) 

Now that you have your Portenta C33 and your Arduino IoT Cloud configured let's create a "Thing" to test the connection between your board and the Arduino IoT Cloud. Navigate into **Things** and select the **CREATE THING** button; give your thing a name. 

![Arduino IoT Cloud "Thing" setup](assets/user-manual-18.png) 

Then, navigate into **Associate Device** and click the **Select Device** button, select your Portenta C33 board and associate it to your "Thing". Then, navigate into **Network** and click the **Configure** button; enter your network credentials. Now you are ready to add some variables to your "Thing"; navigate into **Cloud Variables** and click the **ADD VARIABLE** button. 

![Arduino IoT Cloud "Thing" setup](assets/user-manual-19.png) 

Add one variable with the following characteristics:

- **Name**: `led`
- **Variable type**: `boolean`
- **Variable permission** `Read & Write`
- **Variable update policy**: `On change`

> 

Author: Image added!

You should see the `led` variable on the **Cloud Variables** section. Now, navigate into **Dashboards** and select the **BUILD DASHBOARD** button, this will create a new dashboard; give your dashboard a name. Now, add the following to your widgets to your dashboard:

- **Switch**: name the widget `Switch` and link it to the `led` variable you created before. 
- **LED**: name the widget `led` and link it to the `led` variable you created before. 

Your dashboard should see like the following:

![Arduino IoT Cloud "Thing" setup](assets/user-manual-20.png)

Now, go back to your **Things** and open the "Thing" you just have created. In the "Thing" setup page, navigate into **Sketch**, you should see now an online editor. 

In the generated sketch, you just need to define the `led` pin as an output:

```arduino
// LED_BUILTIN macro access the onboard green LED 
pinMode(LED_BUILTIN, OUTPUT);  

> ```

Reviewer: I know it is obvious, but some of the most newbies will have problems to understand where to put these lines, please improve it put in the more code or text to explain that you have to put that on the setup

Author: I agree; I put the whole setup() function so people stay aware of the code.

In the `onLedChange()` function, you just need to set the LED state with the `led` variable.

```arduino
// led variable synced with the Arduino IoT Cloud dashboard 
digitalWrite(LED_BUILTIN, led);  
```

> The full example code can be found below: 

Reviewer: good idea to have the full code to avoid problem/missconceptions

  The following variables are automatically generated and updated when changes are made to the Thing

  bool led;
*/

#include "thingProperties.h"

void setup() {
  // Initialize serial and wait for port to open:
  Serial.begin(9600);

  // This delay gives the chance to wait for a Serial Monitor without blocking if none is found
  delay(1500); 

  // LED_BUILTIN macro access the onboard green LED 
  pinMode(LED_BUILTIN, OUTPUT);

  // Defined in thingProperties.h
  initProperties();

  // Connect to Arduino IoT Cloud
  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
  
  /*
     The following function allows you to obtain more information
     related to the state of network and IoT Cloud connection and errors
     the higher number the more granular information you’ll get.
     The default is 0 (only errors).
     Maximum is 4
 */
  setDebugMessageLevel(2);
  ArduinoCloud.printDebugInfo();
}

void loop() {
  ArduinoCloud.update();
}

/*
  Since Led is READ_WRITE variable, onLedChange() is
  executed every time a new value is received from IoT Cloud.
*/
void onLedChange()  {
  // led variable synced with the Arduino IoT Cloud dashboard 
  digitalWrite(LED_BUILTIN,led);
}
```

> 

Reviewer: We cannot finish the section this way

We have to finish with:
* Explain the actions the user need to do after it (compile & upload)
* Explain the user what are the final results of these actions to verify everything is working fine (as we said in our last meeting) in this case, explaining that once the code is uploaded to the board, the user should be capable of controlling the LED thought the dashboard and see if it is on or off in real-time.

Reviewer: IMPORTANT

C33 Led is OFF when siwthc is ON and vice-versa, we need to change the code to make the example coherent!!!! @canchebagur

Author: I didn't found another interesting widget we can use... Any idea?

Reviewer: Ok let's do one thing, let keep it this way while I look for alternatives but please do:

"We cannot finish the section this way

We have to finish with:

* Explain the actions the user need to do after it (compile & upload)
* Explain the user what are the final results of these actions to verify everything is working fine (as we said in our last meeting) in this case, explaining that once the code is uploaded to the board, the user should be capable of controlling the LED thought the dashboard and see if it is on or off in real-time.
"

Thanks!!!

Author: Ok, a screenshot and a small section explaining the compile and upload process were added. Also, a small section explaining to the users the final results of those actions.

![Arduino IoT Cloud Setup Device pop-up window](assets/user-manual-17.png) 

Once the Portenta C33 has been configured, let's create a "Thing" to test the connection between your board and the Arduino IoT Cloud. Navigate into **Things** and select the **CREATE THING** button; give your thing a name. 

![Arduino IoT Cloud "Thing" setup](assets/user-manual-18.png) 

Navigate into **Associate Device** and click the **Select Device** button, select your Portenta C33 board and associate it to your "Thing". Then, navigate into **Network** and click the **Configure** button; enter your network credentials. The project is now ready to add some variables to your "Thing"; navigate into **Cloud Variables** and click the **ADD VARIABLE** button. 

> ![Arduino IoT Cloud "Thing" setup](assets/user-manual-19.png) 

Reviewer: ```suggestion
![Add variable button](assets/user-manual-19.png) 

```

- **Public key import**: The public key derived from the generated private key is imported to the secure element with key ID 899.
- **Signature verification**: The example code then verifies the signature using the imported public key and prints a success or failure message accordingly.

## Arduino IoT Cloud 

> The Portenta C33 is fully compatible with the Arduino Cloud IoT, making it easy to develop professional applications simply and quickly. By using the IoT Cloud, you can, for example, monitor sensor data, control your board and actuators connected to it remotely, and update your device's firmware via OTA. 

Reviewer 2: ```suggestion
The Portenta C33 is fully compatible with the Arduino Cloud IoT, which simplifies how professional applications are developed and tracked. By using the IoT Cloud, you can, for example, monitor sensor data, control your board and actuators connected to it remotely, and update your device's firmware over-the-air. 

```

In case it is the first time you are using the Arduino IoT Cloud:

- To use the Arduino IoT Cloud, you need an account. If you do not have an account, create one for free [here](https://cloud.arduino.cc/). 
- To use the Arduino Web Editor or Arduino IoT Cloud, the Arduino Create Agent must be running on your computer. You can install the Arduino Create Agent [here](https://create.arduino.cc/getting-started/plugin/welcome).

> Let's walk through a step-by-step demonstration of how to use your Portenta C33 board with the Arduino IoT Cloud. Log in to your Arduino IoT Cloud account; you should see the following:

Reviewer 2: ```suggestion
Let's walk through a step-by-step demonstration of how to use your Portenta C33 board with the Arduino IoT Cloud. 

Log in to your Arduino IoT Cloud account; you should see the following:

```

![Arduino IoT Cloud initial page](assets/user-manual-13.png) 

> First, associate your Portenta C33 board with your Arduino IoT Cloud. To do this, navigate to **Devices** and then click on the **ADD DEVICE** button:

Reviewer 2: ```suggestion
First, provision your Portenta C33 board on your Arduino IoT Cloud space. To do this, navigate to **Devices** and then click on the **ADD DEVICE** button:

```

![Arduino IoT Cloud Devices page](assets/user-manual-14.png) 

The **Setup Device** pop-up window will appear, navigate into **AUTOMATIC** and select the **Arduino board** option:

![Arduino IoT Cloud Setup Device pop-up window](assets/user-manual-15.png) 

After a while, your Portenta C33 board should be discovered by the Arduino IoT Cloud, as shown below: 

![Arduino IoT Cloud Setup Device pop-up window](assets/user-manual-16.png) 

> Click the **CONFIGURE** button, give your board a name, and select the type of network connection. In this example, we will use a Wi-Fi® connection; you can also use an Ethernet connection with a [Portenta Max Carrier](https://store.arduino.cc/products/portenta-max-carrier), a [Portenta Breakout](https://store.arduino.cc/products/arduino-portenta-breakout), or a custom-made board with an Ethernet connector. Your Portenta C33 board will be configured to communicate securely with the Arduino IoT Cloud. This process can take a while. 

Reviewer 2: ```suggestion
Click the **CONFIGURE** button, give your board a name, and select the type of network connection. In this example, we will use a Wi-Fi® connection; you can also use an Ethernet connection with a [Portenta Max Carrier](https://store.arduino.cc/products/portenta-max-carrier), a [Portenta Breakout](https://store.arduino.cc/products/arduino-portenta-breakout), a [Portenta Vision Shield](https://store.arduino.cc/products/arduino-portenta-vision-shield-ethernet) or a custom-made board with an Ethernet connector. Your Portenta C33 board will be configured to securely communicate with the Arduino IoT Cloud. This process can take a while. 

```

![Arduino IoT Cloud Setup Device pop-up window](assets/user-manual-17.png) 

Once the Portenta C33 has been configured, let's create a "Thing" to test the connection between your board and the Arduino IoT Cloud. Navigate into **Things** and select the **CREATE THING** button; give your thing a name. 

![Arduino IoT Cloud "Thing" setup](assets/user-manual-18.png) 

> Navigate into **Associate Device** and click the **Select Device** button, select your Portenta C33 board and associate it to your "Thing." Then, navigate into **Network** and click the **Configure** button; enter your network credentials. The project is now ready to add some variables to your "Thing"; navigate into **Cloud Variables** and click the **ADD VARIABLE** button. 

Reviewer 2: ```suggestion
Navigate into **Associate Device** and click the **Select Device** button. Select your Portenta C33 board and associate it with your "Thing." Then, navigate into **Network** and click the **Configure** button; enter your network credentials. 

The project is now ready to add some variables to your "Thing"; navigate into **Cloud Variables** and click the **ADD VARIABLE** button. 

```

![Add variable button](assets/user-manual-19.png) 

Add one variable with the following characteristics:

- **Name**: `led`
- **Variable type**: `boolean`
- **Variable permission** `Read & Write`
- **Variable update policy**: `On change`

![Arduino IoT Cloud "Thing" variable setup](assets/user-manual-21.png) 

You should see the `led` variable in the **Cloud Variables** section. Navigate into **Dashboards** and select the **BUILD DASHBOARD** button; this will create a new dashboard; give your dashboard a name. 

Add the following widgets to your dashboard:

- **Switch**: name the widget `Switch` and link it to the `led` variable you created before. 
- **LED**: name the widget `led` and link it to the `led` variable you created before. 

> Your dashboard should see the following:

Reviewer 2: ```suggestion
Your dashboard should look like the following:

```

![Arduino IoT Cloud Dashboard setup](assets/user-manual-20.png)

Go back to your **Things** and open the "Thing" you created. In the "Thing" setup page, navigate into **Sketch**, where you should see the online editor. 

In the generated sketch, define `LED_BUILTIN` pin as an output in the `setup()` function:

```arduino
void setup() {
  // Initialize serial and wait for port to open:
  Serial.begin(9600);
  // This delay gives the chance to wait for a Serial Monitor without blocking if none is found
  delay(1500);

  // LED_BUILTIN macro access the onboard green LED 
  pinMode(LED_BUILTIN, OUTPUT);   

  // Defined in thingProperties.h
  initProperties();

  // Connect to Arduino IoT Cloud
  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
  
  /*
     The following function allows you to obtain more information
     related to the state of network and IoT Cloud connection and errors
     the higher number the more granular information you’ll get.
     The default is 0 (only errors).
     Maximum is 4
 */
  setDebugMessageLevel(2);
  ArduinoCloud.printDebugInfo();
}
```

> In the `onLedChange()` function, you must set the LED state with the `led` variable.

Reviewer 2: I'd suggest to add somewhere that this function is automatically generated for the user by the cloud... Like a low code approach, what do you think? cc @jcarolinares

Author: What about this @martab1994?
![image](https://github.com/arduino/docs-content/assets/20689211/234e0bad-210b-4146-85fc-030b3c23eb7c)

Reviewer: @canchebagur small improvement IMO

In the onLedChange() function, which was generated automatically when the cloud variable led was created, you must vinculate the LED state to the led variable.

Remember that the screenshot that you shared with me does not have the NOT !led so be careful

Author: "Vinculate" sounds weird and seems not to exist or not to be used frequently; change for "associate."

Reviewer 2: Fine for me thanks 👍

  Serial.begin(9600);
  // This delay gives the chance to wait for a Serial Monitor without blocking if none is found
  delay(1500); 

  // Defined in thingProperties.h
  initProperties();

  // Connect to Arduino IoT Cloud
  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
  
  /*
     The following function allows you to obtain more information
     related to the state of network and IoT Cloud connection and errors
     the higher number the more granular information you’ll get.
     The default is 0 (only errors).
     Maximum is 4
 */
  setDebugMessageLevel(2);
  ArduinoCloud.printDebugInfo();
}

void loop() {
  ArduinoCloud.update();
  // Your code here 
}

/*
  Since Led is READ_WRITE variable, onLedChange() is
  executed every time a new value is received from IoT Cloud.
*/
void onLedChange()  {
  digitalWrite(LED_BUILTIN, !led);
}
```

To upload the code to the Portenta C33 from the online editor, click the green **Verify** button to compile the sketch and check for errors, then click the green **Upload** button to program the board with the sketch.

![Uploading a sketch to the Portenta C33 in the Arduino IoT Cloud](assets/user-manual-22.png) 

> Navigate into **Dashboards** again, your board should connect to the Wi-Fi® network you defined before (you can follow the connection process with the online editor integrated Serial Monitor). Your board's green LED should light on or off when the position of the switch is changed. 

Reviewer: Navigate into **Dashboards** again, your board should connect to the Wi-Fi® network you defined before (you can follow the connection process with the online editor integrated Serial Monitor). Your board's green LED should light on or off when the position of the switch changes.

Reviewer: the change is 
![image](https://github.com/arduino/docs-content/assets/7687624/a51133c5-5db5-4990-92cf-3b4aadf76f0e)
